README.md

```diff
@@ -539,6 +541,83 @@ module3 *
 Inclusion and exclusion files can be used together, in that case inclusion
 takes precedence over exclusion.
 
+If you need to set a custom precision for a set of call sites, you can use the functions's instrumentation proposed by verificarlo. Firstly, you need to activate instrumentation by giving the `--inst-func` flag to verificarlo:
+
+```bash
+   $ verificarlo-c main.c -o main --inst-func
+```
+
+Once the file is compilated, you can get information on instrumented functions by producing a profile file: 
+
+```bash
+   $ export VFC_BACKENDS="libinterflop_vprec.so --prec-output-file=output.txt" ./main
+```
+
+In this file you can see information on the functions and on floating point arguments with the following structure: 
+
+```
+file/name_line  precision_binary64  range_binary64  precision_binary32  range_binary32  nb_inputs nb_outputs  nb_calls
+input:  type  precision   range
+...
+input:  type  precision   range
+output: type  precision   range
+```
+
+Where:
+  - `file` is the file which containts the call of the function
+  - `name` is the name of the called function
+  - `line` is the line where the function is called in the source code
+  - `precision_binary64` control the length of the mantissa for a 64 bit float in internal operations
+  - `range_binary64` control the length of the exponent for a 64 bit float in internal operations
+  - `precision_binary32` control the length of the mantissa for a 32 bit float in internal operations 
+  - `range_binary32` control the length of the exponent for a 64 bit float in internal operations
+  - `nb_inputs` is the number of floating point inputs intercepted
+  - `nb_outputs` is the number of floating point outputs intercepted
+  - `nb_calls` is the number of calls to the function from this site
+  - `type` is the type of the argument (0 = float and 1 = double)
+  - `precision` is the length of the mantissa for this argument
+  - `range` is the length of the exponent for this argument
+
+Only floating point arguments are managed by vprec, so for example this code: 
+
+```c
+double print(int n, double a, double b) {
+  for (int i = 0; i < n; i++)   
+    printf("%lf %lf\n", a, b);
+  return a + b;
+}
+
+...
+
+double res = print(2, 3.5);
+```
+will produce this set of informations:
+
+```
+main.c/print_20  52  11  23  8  2 1 1
+# a 
+input:  1  52   23
+# b
+input:  1  52   23
+# res
+output: 1  52   23
+```
+
+You are now able to customize the length of the mantissa/exponent for each argument but also to set the internal precision for floating point operations in a function compilated with verificarlo.
+To do that you can change the desired field in the profile file and using it as an input with the following command:
```

Comment:
using it as an input -> use it as an input

Reply:
typos are corrected